### ISSUE TITLE
Alter `MDAnalsysi.base.AnanlysisBase`

### ISSUE THREAD

**Issue by PicoCentauri** [MDAnalysis/mdacli#35]
I know that @joaomcteixeira is not a fan of OOPing things but I will post this here anyway ;).
I thought about altering the base class with two new methods/attributes. A `self.doc_dict` attribute: a dictionary combining the docstring information and the callable signature. In principle, it is our 

https://github.com/MDAnalysis/mdacli/blob/78fa3f2fff31fb7f9acf71ee8811ab902369e947/src/mdacli/utils.py#L62

as a class method.

And a `self.save_results()` method based on our general approach could be added if no such a function exists. 

If we dynamically alter the analysis class we are parsing this could help to improve our workflow and will help users/developers that try to use our python implementation.

**Comment by joaomcteixeira** [comment]
I am fan of OOP where OOP adds value :laughing: 

I see, you want to polymorph the Analysis class possible save_results() with our own saving engine so that later in the CLI we only need to do `.save_results()` and avoid the try:catch block. Is that it? It can be a good application.

`self.doc_dict` is bad. Better have a `self.doc` and `self.sig`.

We need to do this only after #17 and #28 are merged, otherwise become too messy. :+1:

**Comment by PicoCentauri** [comment]
Yes first of all to avoid the try/catch block. But this could also help us to remove the "reparsing" in 

https://github.com/MDAnalysis/mdacli/blob/6334ac40bc30bb1ff306f56c40e34042cf6be1fc/src/mdacli/cli.py#L277-L278

 and helps solving #32.
